packages/cli/src/commands/trace.ts

```diff
@@ -45,6 +45,15 @@ export async function trace({
   // will call `trace_transaction`, and decode as much data from the trace
   // as possible, the same way that an error occurs
 
+  if (providerUrl) {
+    // get chain id from provider
+    const publicClient = viem.createPublicClient({
+      transport: viem.http(providerUrl),
+    });
+
+    chainId = await publicClient.getChainId();
+  }
```

Comment:
if user specifies both chain id and provider url and the chain id doesnt match, then they are probably making a mistake. rather than moving forward and assuming everything is fine, we should emit an error ssaying the chain id doesn't match.